contracts/liquidity_pool.cairo

```diff
@@ -764,6 +790,124 @@ func _get_value_of_pool_position{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*,
 }
 
 
+@view
+func get_user_poolinfo{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+} (user: Address) -> (
+    upoolinfo_len: felt,
+    upoolinfo: UserPoolInfo*
+) {
+    alloc_locals;
+    let (lptoken_addrs_len: felt, lptoken_addrs: Address*) = get_all_lptoken_addresses();
+    let (res: UserPoolInfo*) = alloc();
+    let (upoolinfo_len: felt) = map_and_filter_address_to_userpoolinfo(user, lptoken_addrs, lptoken_addrs_len, res);
+    return (upoolinfo_len, res);
+}
+
+func map_and_filter_address_to_userpoolinfo{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+} (user_addr: Address, lptoken_addrs: Address*, lptoken_addrs_len: felt, upoolinfo: UserPoolInfo*)
+    -> (upoolinfo_len: felt){
+    if (lptoken_addrs_len == 0){
+        return (0,);
+    }
+    let val = get_userpoolinfo(user_addr, lptoken_addrs[0]);
+    if (val.value.low == 0){
+        return map_and_filter_address_to_userpoolinfo(user_addr, lptoken_addrs + 1, lptoken_addrs_len - 1, upoolinfo);
+    }
+    assert [upoolinfo] = val;
+    let (retval: felt) = map_and_filter_address_to_userpoolinfo(user_addr, lptoken_addrs + 1, lptoken_addrs_len - 1, upoolinfo + UserPoolInfo.SIZE);
+    let retval_adj = retval + 1;
+    return (retval_adj,);
+}
+
+// func get_value_of_capital_in_lpool{
+//     syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+// } () -> (
+// ) {
+
+// }
+
+@view
+func get_all_poolinfo{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+} () -> (
+    poolinfo_len: felt,
+    poolinfo: PoolInfo*
+) {
+    alloc_locals;
+    let (lptoken_addrs_len: felt, lptoken_addrs: Address*) = get_all_lptoken_addresses();
+    // map address->poolinfo via get_poolinfo
+    let (res: PoolInfo*) = alloc();
+    map_address_to_poolinfo(lptoken_addrs, res, lptoken_addrs_len, 0);
+    //let (array : PoolInfo*) = alloc();
+    //let pool = Pool(0, 0, 0);
+    //let poolinfo = PoolInfo(pool, 0, 0, 0, 0);
+    //assert array[0] = poolinfo;
+    return (lptoken_addrs_len, res);
```

Comment:
Are you sure it will work with the `lptoken_addrs_len` even though the len is much bigger?